src/dash-playlist-loader.js

```diff
@@ -670,6 +670,26 @@ export default class DashPlaylistLoader extends EventTarget {
     }
   }
 
+  /**
+   * Given a new manifest, update our pointer to it and update the srcUrl based on the location elements of the manifest, if they exist.
+   *
+   * @param {Object} updatedManifest the manifest to update to
+   */
+  updateMainManifest_(updatedManifest) {
+    this.master = updatedManifest;
+
+    // if locations isn't set or is an empty array, exist early
+    if (!this.master.locations || this.master.locations && !this.master.locations.length) {
```

Comment:
should this just be `!this.master.locations || !this.master.locastions.length`

Reply:
Let me check. I thought I tried this and it was showing an error but I may have tried it with `&&`?

Reply:
yeah, I think I had `&&` locally. with `||` seems to work.